src/main/java/wooteco/subway/line/LineController.java

```diff
@@ -0,0 +1,63 @@
+package wooteco.subway.line;
+
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+public class LineController {
+
+    private final LineService lineService;
+
+    public LineController(LineService lineService) {
+        this.lineService = lineService;
+    }
+
+    @PostMapping("/lines")
+    public ResponseEntity<LineResponse> createLine(@RequestBody LineRequest lineRequest) {
```

Comment:
```suggestion
    @PostMapping("/lines")
    public ResponseEntity<LineResponse> createLine(@RequestBody LineRequest lineRequest) {
        LineResponse lineResponse = lineService.createLine(lineRequest);
        return ResponseEntity.created(URI.create("/lines/" + createdLine.getId()))
                .body(lineResponse);
    }
```
아직도 컨트롤러에 많은 로직이 있네요
더 간결하게 만들 수 있지 않을까요

Reply:
다른 메서드도 같이 점검해보아요

Reply:
혹시 lineRequest를 분해해서 Line객체를 만드는 내용을 정리해보라고 하신 걸까요? 

```java
// LineRequest
    public Line lineWithoutId() {
        return Line.of(name, color);
    }
```
❓ LineRequest에서 가져온 정보로 Line객체를 만들어주는 메소드를 만들었는데 이런 방식을 혹시 의도하신 걸까요??

Reply:
컨트롤러에서 Line, LineResponse를 직접 만들지 말고 서비스 레이어에서 처리하라는 뜻입니다.
dto로 변환하는 일도 컨트롤러에서 처리할 일이 아닙니다.